### PR TITLE
Consolidate Nova integration APIs with External Data APIs

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This PR is big in terms of lines of code but there's not much going on. I removed `NovaLuancher**` types and replaced them with more generic `ExternalData` types. The large line count is due to centralizing all queries in a single `ExternalDataDao` and moving related tests to `ExternalDataDaoTeset`.

The only real change is that the episode data is now modeled more nicely. DAO no longer exposes a type with `isPodcastEpisode` property but instead does it through a sealed interface.

## Testing Instructions

Smoke test Nova Launcher integration. App link: p1718205443795369-slack-C028JAG44VD

## Checklist
- [x] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [x] ~Any jetpack compose components I added or changed are covered by compose previews~
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ExternalPodcastOrUserEpisode</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ExternalPodcastMap</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ExternalDataManagerImpl</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by geekygecko — APPROVED** [review]
LGTM. I'm sorry. I did ask some questions, but then I realised that the class had just moved. Feel free to ignore them.

File: modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/ExternalDataDao.kt [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+          (SELECT MIN(podcast_episodes.published_date) FROM podcast_episodes WHERE podcasts.uuid IS podcast_episodes.podcast_id) AS initial_release_timestamp, 
+          (SELECT MAX(podcast_episodes.published_date) FROM podcast_episodes WHERE podcasts.uuid IS podcast_episodes.podcast_id) AS latest_release_timestamp,
+          (SELECT MAX(podcast_episodes.last_playback_interaction_date) FROM podcast_episodes WHERE podcasts.uuid IS podcast_episodes.podcast_id) AS last_used_timestamp
+        FROM 
+          podcasts 
+        WHERE 
+          podcasts.subscribed IS NOT 0
+        ORDER BY
+          -- Order by oldest to newest date added
+          CASE WHEN :sortOrder IS 0 THEN IFNULL(podcasts.added_date, 9223372036854775807) END ASC,
```
[the remark this answers is not in the document]
Comment (MiSikora) [reply modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/ExternalDataDao.kt:33]: Yes, this is the max `INTEGER` value. I'll append this information to the comment.

File: modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/ExternalDataDao.kt [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+          curated_podcast.*
+        FROM 
+          curated_podcasts as curated_podcast 
+          LEFT JOIN podcasts AS podcast ON podcast.uuid = curated_podcast.podcast_id 
+        WHERE 
+          IFNULL(podcast.subscribed, 0) IS 0
+          OR curated_podcast.list_id IS '${CuratedPodcast.FEATURED_LIST_ID}'
+        """,
+    )
+    abstract suspend fun _getCuratedPodcasts(): List<CuratedPodcast>
```
[the remark this answers is not in the document]
Comment (MiSikora) [reply modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/ExternalDataDao.kt:91]: The public method using it had the same name but I've changed it during development of this task. I used the same "trick" that is used for private backing fields with `_` in the name. I'll fix it.

**Pull request merged**